docs/operation/upgrade.rst

```diff
@@ -0,0 +1,110 @@
+      a successful upgrade.
+    - The output of the below-provided commands solely depends on the particular
+      MetalK8s setup you have in place.
+
+.. warning::
+    For any cluster upgrade from MetalK8s v2.0 to v2.1 and on-ward, **Kubelet**
+    must be upgraded before making any upgrades to the salt-master. Once this
+    is done, an upgrade is guaranteed to proceed in a seamless manner.
+
+Supported Versions
+******************
+
+This guide currently supports MetalK8s upgrades for the following versions.
+Please refer to the release notes for more information.
+
+    - Upgrade 2.0.0 to 2.1.0
+    - Upgrade 2.0.0 to 2.2.0
+    - Upgrade 2.1.0 to 2.2.0
+
+
+Upgrade Steps
+*************
+This section describes the logical steps for Upgrading MetalK8s.
+
+#. Connect to the :term:`Bootstrap node`.
+
+#. Download the new `.ISO` from `Sources <http://scality.com>`_.
+
+#. Edit the `bootstrap.yaml` file located in `/etc/metalk8s/bootstrap.yaml`
+   then add the new MetalK8s ISO mount point.
+
+   .. code::
+
+      products:
+        metalk8s:
+        - /srv/scality/metalk8s-2.0.0-dev
+        - /srv/scality/metalk8s-<new_version>     <=== New line to add
+
+
+#. Mount the new MetalK8s ISO.
+
+    ::
+
+        [root@bootstrap vagrant]# mkdir -p /srv/scality/metalk8s-<new_version>/
+
+    ::
+
+        [root@bootstrap vagrant]# mount metalk8s.iso /srv/scality/metalk8s-<new_version>/
+
+#. Verify that the current MetalK8s version is mounted.
+
+    ::
+
+        salt-call state.sls metalk8s.products.mounted saltenv=metalk8s-<current_version>
+
+            Summary for local
+            ------------
+            Succeeded: 5 (changed=1)
+            Failed:    0
+            ------------
+            Total states run:     5
+
+#. Advertise the new version and make it available to the entire cluster.
+
+    ::
+
+        salt-call --local state.sls metalk8s.products saltenv=metalk8s-<new_version> pillar="{'metalk8s': {'endpoints': $(salt-call --out txt pillar.get metalk8s:endpoints | cut -c 8-)}}"
+
+#. From the :term:`Bootstrap node`, connect to the :term:`Salt Master`.
+
+    ::
+
+        kubectl -n kube-system exec -it salt-master-bootstrap bash
+
+    .. warning::
+        The next few commands must be executed only on the :term:`Salt Master`.
+        Before proceeding, make sure that this is verified.
+
+#. From the :term:`Salt Master`, launch the upgrade procedure.
+
+    ::
+
+        salt-run state.orchestrate metalk8s.orchestrate.upgrade saltenv=metalk8s-<new_version> pillar="{'orchestrate': {'dest_version': '<new_version>'}}"
+
+    .. note::
+        By default, the orchestrate mechanism will upgrade pods and cluster
+        components on a one by one basis.
+
+#. Verify that the upgrade was successful.
+
+   Verify the currently installed version of MetalK8s.
+
+::
+    Command to check if the version has been upgraded properly
+
+Upgrade Considerations
+**********************
+
+Teddy??
+- What to expect during upgrades?
```

Comment:
I don't know what we should check. Maybe at least the kubelet versions with a `kubectl get nodes`